lib/collection/proxy.js

```diff
@@ -0,0 +1,206 @@
+var _ = require('../util').lodash,
+    Property = require('./property').Property,
+    Url = require('./url').Url,
+
+    E = '',
+    SELECTALL = '*',
+    HTTPS = 'https',
+    HTTP = 'http',
+    DEFAULT_PORT = '80',
+    regexes = {
+        proxyStringSanitizer: /(PROXY)|(DIRECT)|\[|\]|\s/g
+    },
+
+    Proxy;
+
+
+/**
+* You can either provide the options object alone to the constructor or the ProxyString and the UrlString
+*
+* The following is the object structure accepted as constructor parameter while calling `new Proxy(...)`. It is
+* also the structure exported when {@link Property#toJSON} or {@link Property#toObjectResolved} is called on a
+* Proxy instance.
+* @typedef Proxy~definition
+*
+* @property {String=} [url] The url for which the proxy needs to be configured @default would '*' means select all url.
+* @property {String=} [protocol] The proxy server protocol by @default it will be taken as 'http'
+* @property {String=} [host] The proxy server host name @default would be '' [Empty String] represents no proxy
+* @property {String=} [port] The proxy server port @default would be '80'
+* @property {Boolean=} [tunnel] The option to provide whether tunneling needs to be done while proxying the request,
+* @default would be true.
+*
+* @example <caption>JSON definition of an example proxy object</caption>
+* {
+*     url: 'https://example.com',
+*     protocol: 'http',
+*     host: 'proxy.com',
+*     port: 88,
+*     tunnel: true
+* }
+*
+* Incase of ProxyString format
+* @param {String=} [options] supported formats return by the PAC files as provided here
+* (https://web.archive.org/web/20070602031929/http://wp.netscape.com/eng/mozilla/2.0/relnotes/demo/proxy-live.html)
+* @param {String=} [url] The url for which the proxy needs to be configured @default would '*' means select all url.
+
+* @example
+* var proxyObj = new Proxy('PROXY [https://proxy.com]:88', 'http://example.com')
+* {
+*     url: 'http://example.com',
+*     protocol: 'http',
+*     host: 'proxy.com',
+*     port: '88',
+*     tunnel: true
+* }
+*/
+_.inherit((
+    Proxy = function PostmanProxy (options, url) {
+        // this constructor is intended to inherit and as such the super constructor is required to be executed
+        Proxy.super_.call(this, options);
+
+        // Assign defaults before proceeding
+        _.assign(this, /** @lends Variable.prototype */ {
+            /**
+             * The url for which the proxy has been associated with.
+             * @type {String}
+             */
+            url: SELECTALL,
+
+            /**
+             * @type {String}
+             * @default 'http'
+             */
+            protocol: HTTP,
+
+            /**
+             * @type {String}
+             * @default ''
+             */
+            host: E,
+
+            /**
+             * @type {String}
+             * @default '80'
+             */
+            port: DEFAULT_PORT,
```

Comment:
There is usually no default port for proxies, right? 
If no port is provided, it should be determined from the URL, and set to 443 or 80 depending on whether https is used or not.

Regardless, I think it's best to not assign defaults in the constructor, and assign defaults only in the `.update()` method.